test/src/apps/signal_c/signal_fd.c

```diff
@@ -0,0 +1,262 @@
+		.events = POLLIN,
+	};
+
+	TEST_SUCC(raise(SIGUSR1));
+	TEST_RES(poll(&pfd, 1, -1), pfd.revents == POLLIN);
+
+	TEST_SUCC(close(sfd));
+	sfd = CHECK(signalfd(-1, &mask, SFD_CLOEXEC | SFD_NONBLOCK));
+	pfd.fd = sfd;
+	pfd.revents = 0;
+	TEST_RES(poll(&pfd, 1, -1), pfd.revents == POLLIN);
+	struct signalfd_siginfo fdsi;
+	TEST_RES(read(sfd, &fdsi, sizeof(fdsi)),
+		 _ret == sizeof(fdsi) && fdsi.ssi_signo == SIGUSR1);
+}
+END_TEST()
+
+FN_TEST(kill_thread)
+{
+	pid_t pid = TEST_SUCC(getpid());
+	TEST_SUCC(tgkill(pid, pid, SIGUSR1));
+
+	struct signalfd_siginfo fdsi;
+	TEST_RES(read(sfd, &fdsi, sizeof(fdsi)),
+		 _ret == sizeof(fdsi) && fdsi.ssi_signo == SIGUSR1);
+}
+END_TEST()
+
+FN_TEST(kill_thread_and_process)
+{
+	pid_t pid = TEST_SUCC(getpid());
+
+	TEST_SUCC(tgkill(pid, pid, SIGUSR1));
+	TEST_SUCC(kill(pid, SIGUSR1));
+
+	struct signalfd_siginfo fdsi[2];
+	TEST_RES(read(sfd, fdsi, 2 * sizeof(struct signalfd_siginfo)),
+		 _ret == 2 * sizeof(struct signalfd_siginfo) &&
+			 fdsi[0].ssi_signo == SIGUSR1 &&
+			 fdsi[1].ssi_signo == SIGUSR1);
+}
+END_TEST()
+
+FN_TEST(kill_process_and_thread)
+{
+	pid_t pid = TEST_SUCC(getpid());
+
+	TEST_SUCC(kill(pid, SIGUSR1));
+	TEST_SUCC(tgkill(pid, pid, SIGUSR1));
+
+	struct signalfd_siginfo fdsi[2];
+	TEST_RES(read(sfd, fdsi, 2 * sizeof(struct signalfd_siginfo)),
+		 _ret == 2 * sizeof(struct signalfd_siginfo) &&
+			 fdsi[0].ssi_signo == SIGUSR1 &&
+			 fdsi[1].ssi_signo == SIGUSR1);
+}
+END_TEST()
+
+void *thread_func(void *arg)
+{
+	CHECK(close(sfd));
+	sfd = CHECK(signalfd(-1, &mask, SFD_CLOEXEC));
+	sleep(2);
+	pid_t pid = CHECK(getpid());
+	CHECK(tgkill(pid, pid, SIGUSR1));
+}
+
+FN_TEST(tgkill_other_thread)
+{
+	pthread_t tid;
+	TEST_SUCC(pthread_create(&tid, NULL, thread_func, NULL));
+	sleep(1);
+	struct signalfd_siginfo fdsi;
+	TEST_RES(read(sfd, &fdsi, sizeof(fdsi)),
+		 _ret == sizeof(fdsi) && fdsi.ssi_signo == SIGUSR1);
+	pthread_join(tid, NULL);
+}
+END_TEST()
+
+void *thread_func2(void *arg)
+{
+	int *pipefd = (int *)arg;
+	pid_t tid = CHECK(gettid());
+	CHECK(write(pipefd[1], &tid, sizeof(tid)));
+	sleep(1);
+
+	char buf[1];
+	CHECK_WITH(read(pipefd[0], buf, sizeof(buf)),
+		   _ret == 1 && buf[0] == 'a');
+
+	sigset_t sigset;
+	CHECK_WITH(sigprocmask(SIG_BLOCK, NULL, &sigset),
+		   sigisemptyset(&sigset));
+	CHECK_WITH(sigpending(&sigset), sigisemptyset(&sigset));
+}
+
+FN_TEST(blocking_syscall_dequeue_ignored_signals)
+{
+	int pipefds[2];
+	TEST_SUCC(pipe(pipefds));
+
+	sigset_t mask2;
+	sigemptyset(&mask2);
+	TEST_SUCC(sigprocmask(SIG_SETMASK, &mask2, NULL));
+
+	pthread_t tid;
+	TEST_SUCC(pthread_create(&tid, NULL, thread_func2, (void *)pipefds));
+
+	TEST_SUCC(signal(SIGUSR2, SIG_IGN));
+
+	sigaddset(&mask2, SIGUSR2);
+	TEST_SUCC(sigprocmask(SIG_SETMASK, &mask2, NULL));
+
+	pid_t real_tid;
+	TEST_SUCC(read(pipefds[0], &real_tid, sizeof(real_tid)));
+
+	TEST_SUCC(sleep(2));
+	TEST_SUCC(tgkill(getpid(), real_tid, SIGUSR2));
+	TEST_SUCC(kill(getpid(), SIGUSR2));
+	TEST_SUCC(sleep(1));
```

Comment:
Be consistent. If all the previous tests do not `TEST_SUCC(sleep(x))`, then this test should also have no `TEST_SUCC` on `sleep(x)`?